api/handlers/handlers.go

```
@@ -110,66 +120,83 @@ func (h *ModelHandler) CreateModel() {
 // @Produce      json
 // @Param        model  body  apiTypes.CausalDecisionModel  true  "Causal Decision Model Payload"
 // @Success      201 {object} apiTypes.CausalDecisionModel "Created model"
+// @Failure      400 {object} gin.H "Bad Request"
+// @Failure      409 {object} gin.H "Conflict: Model with same UUID already exists"
 // @Failure      500 {object} gin.H "Internal Server Error"
 // @Router       /v0/models/ [post]
 func (h *ModelHandler) UploadModel(c *gin.Context) {
 	var uploadedModel apiTypes.CausalDecisionModel
 
-	c.ShouldBindJSON(&uploadedModel)
-
-	transacation := h.DB.Begin()
+	if err := c.ShouldBindJSON(&uploadedModel); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"Error": err.Error()})
+		return
+	}
 
-	//if what is passed in doesnt aleayd have meta data we need to create meta data
+	// Get the uploaded model's meta info.
+	meta := uploadedModel.Meta
 
-	err := transacation.Create(&uploadedModel.Meta).Error
+	// Ensure no other model with the same UUID exists.
+	var existingMeta apiTypes.Meta
+	if err := h.DB.Where("uuid = ?", meta.UUID).First(&existingMeta).Error; err == nil {
+		// If there wasn't an error (error is nil), then a meta with the same UUID exists
```

So if there isn't an error, we need to throw an error?